community/server/src/main/java/org/neo4j/server/rest/dbms/UserService.java

```diff
@@ -136,4 +146,16 @@ public Response setPassword( @PathParam("username") String username, @Context Ht
         return output.ok();
     }
 
+    private AuthSubject getSubjectFromPrincipal( Principal principal )
+    {
+        if ( principal instanceof DelegatingPrincipal )
+        {
+            AccessMode mode = ((DelegatingPrincipal) principal).getAccessMode();
+            if ( mode instanceof AuthSubject )
+            {
+                return (AuthSubject) mode;
+            }
+        }
+        return null;
```

Comment:
There was already a static AuthorizedRequestWrapper.getAccessModeFromUserPrincipal(). It was hidden in there to hide as much of this dirtiness as possible from the outside world.